src/azul/plugins/metadata/hca/transform.py

```diff
@@ -989,8 +1070,55 @@ def transform(self) -> Iterable[Contribution]:
                                     map(self._sequencing_process, visitor.sequencing_processes.values())
                                 ),
                                 projects=[self._project(project)])
+                # Supplementary file matrices provide stratification values that
+                # need to be reflected by inner entities in the contribution.
+                if isinstance(file, api.SupplementaryFile):
+                    if Submitter.category_for(file) is not None:
+                        contents = {
+                            **contents,
+                            **self.matrix_stratification_values(file)
+                        }
                 yield self._contribution(contents, file.document_id)
 
+    def matrix_stratification_values(self,
+                                     file: api.SupplementaryFile) -> JSON:
+        points = {
+            dimension: values.split(',')
+            for dimension, values in (point.split('=')
+                                      for stratum in file.json['file_description'].split('\n')
```

Comment:
Extract `file.json['file_description']` into a variable and move the parsing code into `contributor_matrices.py` for encapsulation. The function signature should be `def parse_stratification(strata:str) -> Mapping[…]` with … reflecting the exact structure of the return value.  Add doctests to that function.